File: Jan/Week 2/LeetCode/problem_6.cpp
//  Day 13 Problem No 26

// 443. String Compression => https://leetcode.com/problems/string-compression/description/?envType=study-plan-v2&envId=leetcode-75

class Solution {
public:
    int compress(vector<char>& chars) {
        int w = 0;
        int r = 0;
        while(r < chars.size()){
            char currentChar = chars[r];
            int count = 0;
            while(r < chars.size() && currentChar == chars[r]){
                count++;
                r++;
            }
            chars[w++] = currentChar;
            if(count > 1){
                for(char c:to_string(count)){
                    chars[w++] = c;
                }
            }

        }
        return w;
    }
};

File: Jan/Week 3/LeetCode/problem_6.py
# Day 20 Problem No 43

# 724. Find Pivot Index => https://leetcode.com/problems/find-pivot-index/description/?envType=study-plan-v2&envId=leetcode-75

class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total = sum(nums)
        l_sum = 0

        for i in range(len(nums)):
            if l_sum == total - l_sum - nums[i]:
                return i
            else:
                l_sum += nums[i]
        return -1